app/models/manageiq/providers/hawkular/middleware_manager/middleware_deployment.rb

```diff
@@ -0,0 +1,4 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::MiddlewareDeployment < MiddlewareDeployment
```

Comment:
After reviewing other providers. 
The design uses a generic class of the provider that is extended in the provider implementation.
Also, after implementing MetricsCapture workers, the design classes also expects that this class is extended in the provider.
Also it has sense if we want to add additional middleware providers in the future.